### PR TITLE
Functionality to add bonus scores on tag updates

### PR DESCRIPTION
**Pull request by art4ride**

### CODE REVIEW

File: imports/api/eligibleClients/methods.js [hunk 1]
```diff
@@ -23,6 +23,21 @@ Meteor.methods({
     return hc.api('house-matching-v2').getEligibleClientsPage(pageNumber, pageSize, sort, order);
   },
 
+  updateBonusScore(clientId, bonusScore) {
```
Comment (pgorecki) [inline imports/api/eligibleClients/methods.js]: We mostly have string-based method names i.e. 'module.function', so yours should be 'eligibleClients.updateBonusScore'

File: imports/api/eligibleClients/methods.js [hunk 2]
```diff
@@ -23,6 +23,21 @@ Meteor.methods({
...
+    logger.info(`METHOD[${this.userId}]: updateScores(${Array.isArray(clientId)}, ${clientId})`); // eslint-disable-line max-len
+    check(clientId, Match.OneOf(String, [String])); // eslint-disable-line new-cap
+    if (!Roles.userIsInRole(this.userId, ClientsAccessRoles)) {
+      throw new Meteor.Error(403, 'Forbidden');
+    }
+
+    const hc = HmisClient.create(this.userId);
+    // get client details
+    const eligibleClient = hc.api('house-matching').getEligibleClient(clientId);
```
Comment (pgorecki) [inline imports/api/eligibleClients/methods.js]: what if clientId is an array? shouldnt we have clientId.forEach(...) is this case?

File: imports/api/eligibleClients/methods.js [hunk 3]
```diff
@@ -23,6 +23,21 @@ Meteor.methods({
...
+    eligibleClient.bonusScore = bonusScore;
+    eligibleClient.totalScore = eligibleClient.bonusScore + eligibleClient.surveyScore / 1;
```
Comment (pgorecki) [inline imports/api/eligibleClients/methods.js]: What if you call this method twice in a row? shouldn't you remove the previous bonus from total first and then apply a new bonus?

File: imports/ui/clients/clientTagList.js [hunk 4]
```diff
@@ -3,6 +3,15 @@ import ClientTagList from '/imports/ui/components/tags/ClientTagList.js';
...
 import { ClientTags } from '/imports/api/tags/clientTags.js';
 import { setAppContext } from '/imports/ui/app/appContext';
+import { getActiveTagsForDate } from '/imports/api/tags/tags';
+
+const getBonusScore = () => {
+  const clientTags = ClientTags.find().fetch();
+  const now = moment().format('YYYY-MM-DD');
+  const activeTags = getActiveTagsForDate(clientTags, now);
+  const activeTagScores = activeTags.map((tag) => tag.score);
+  return _.max(activeTagScores);
```
Comment (pgorecki) [inline imports/ui/clients/clientTagList.js]: don't forget to change this one :)

File: imports/api/eligibleClients/methods.js (hunk 3 above)
Comment (art4ride) [reply imports/api/eligibleClients/methods.js]: Bonus is recalculated on every call. Ignoring previous value

File: imports/ui/components/tags/TagItem.js [hunk 5]
```diff
@@ -1,19 +1,53 @@
...
-    this.removeTag = this.removeTag.bind(this);
+    this.removeButton = this.removeButton.bind(this);
+    this.editTag = this.editTag.bind(this);
+    this.editButton = this.editButton.bind(this);
+    this.saveTag = this.saveTag.bind(this);
   }
 
-  removeTag() {
-    return () => this.props.removeTagHandler(this.props.data.id);
+  editTag() {
```
Comment (pgorecki) [inline imports/ui/components/tags/TagItem.js]: better name would be toggleEditTag or passing an argument to set the edit state

File: imports/ui/components/tags/TagItem.js [hunk 6]
```diff
@@ -22,33 +56,52 @@ class TagItem extends Component {
...
+    </span>);
   }
   readOnlyField() {
     const { data: { name, score } } = this.props;
 
-    return (<div>{name}:&nbsp;{score}</div>);
+    return (<span>{name}:&nbsp;{score}</span>);
+  }
+
+  editButton() {
```
Comment (pgorecki) [inline imports/ui/components/tags/TagItem.js]: change name to renderEditButton

File: imports/ui/components/tags/TagItem.js [hunk 7]
```diff
@@ -22,33 +56,52 @@ class TagItem extends Component {
...
         onChange={(value) => this.handleChange('score', value)}
+        defaultValue={this.state.score}
+        placeholder="Tag score"
       />
-    </div>);
+      <a style={{ margin: '5px' }}><i className="fa fa-save" onClick={this.saveTag}></i></a>
+      {/* this.editButton() */}
+    </span>);
   }
   readOnlyField() {
```
Comment (pgorecki) [inline imports/ui/components/tags/TagItem.js]: change name to renderReadOnlyField

File: imports/ui/components/tags/TagItem.js [hunk 8]
```diff
@@ -22,33 +56,52 @@ class TagItem extends Component {
   }
 
   editableField() {
```
Comment (pgorecki) [inline imports/ui/components/tags/TagItem.js]: change name to renderEditableField

File: imports/ui/components/tags/TagItem.js [hunk 9]
```diff
@@ -1,19 +1,53 @@
...
+    const { name, score } = this.state;
+    this.props.updateTagHandler(this.props.data.id, name, score);
+    this.editTag();
+  }
+
+  // removeTag() {
+  //   return this.props.removeTagHandler(this.props.data.id);
+  // }
+
+  removeButton() {
```
Comment (pgorecki) [inline imports/ui/components/tags/TagItem.js]: change name to renderRemoveButton

**Comment by art4ride** [comment]
@pgorecki namings resolved

**Pull request merged**